src/components/Sidebar/Tabs/EditTab.tsx

```diff
@@ -628,3 +642,47 @@ function extractPillProps(spec: GraphSpec) {
   });
   return Object.values(pillsByField).flat();
 }
+
+/**
+ * Determines whether a given mark is applicable to a GraphSpec.
+ * @param mark The new mark option to be validated.
+ * @param spec Spec receiving the new mark
+ * @returns true if the mark is valid.
+ */
+function validateMarkChange(mark: VegaMark, spec: GraphSpec): boolean {
+  const xAxisType = spec.encoding['x']?.type;
+  const yAxisType = spec.encoding['y']?.type;
+
+  const isAxisCombination = (types: (VegaColumnType | undefined)[]) => {
+    if (types.length > 1) {
+      return (
+        (types[0] === xAxisType && types[1] === yAxisType) ||
+        (types[1] === xAxisType && types[0] === yAxisType)
+      );
+    } else if (types.length === 1) {
+      return xAxisType === types[0] || yAxisType === types[0];
+    } else return false;
+  };
+
+  switch (mark) {
+    case 'line':
+      return isAxisCombination(['quantitative', 'quantitative']);
+    case 'errorband':
+      return isAxisCombination(['ordinal', 'quantitative']);
+    case 'errorbar':
+      return isAxisCombination(['quantitative', 'nominal']);
+    case 'tick':
+      return (
+        isAxisCombination([undefined, 'quantitative']) ||
+        isAxisCombination(['nominal', 'quantitative'])
```

Comment:
Shouldn't tick mark work with quantitative and quantitive?

Reply:
It could technically work, but idk if it would be useful. Wouldn't a scatterplot make more sense in that case?

Reply:
Yea, scatterplot makes so much more sense to use in this case, but I don't think we should leave something out because it's something that users are not likely to use unless adding it gives us so much trouble.